### ISSUE TITLE
ONE OF contraint not working in template form

### ISSUE THREAD

**Issue by rachellyne** [intermine/intermine#2044]
I updated this template to use a ONE OF constraint: http://www.flymine.org/flymine/template.do?name=Stage_FlyFishBDGP&scope=all. 

It works ok from the query builder but when you try and run it from the template form it turns the constraint into this:

 ONE OF [[Ljava.lang.String;@4e9172f3]  [Remove this constraint]   [Edit this constraint]  (B)

**Comment by julie-sullivan** [comment]
works fine if you "edit template", does not work as expected when you "edit query". I suppose constraints are handled differently in the two.

**Comment by julie-sullivan** [comment]
When I try to use that template, it works on SOME stages but not all. When I use the first stage, I get an error about whitespace. (see below)

1. should the data parser parse out the whitespace (yes)
2. should the ONE OF constraints be updated to not be so delicate (also yes)

```
Caused by: javax.xml.stream.XMLStreamException: Value in MultiValue starts or ends with whitespace - this query cannot be represented in XML
	at org.intermine.pathquery.PathQueryBinding.marshalPathQueryConstraints(PathQueryBinding.java:212)
	at org.intermine.pathquery.PathQueryBinding.doMarshal(PathQueryBinding.java:122)
	... 56 more
```

**Comment by julie-sullivan** [comment]
I am pretty sure this change is related to our problem: https://github.com/intermine/intermine/commit/a287177b75f05678322711cc361e44c094eb0ad3

**Comment by julie-sullivan** [comment]
Fixed by https://github.com/julie-sullivan/intermine/commit/1f2aa0a644b4beb6867772b170a9988389ae062a

**Comment by julie-sullivan** [comment]
Also, @yochannah you updated the tutorial to include this new feature. Is that right? Can you make that live too?

**Comment by julie-sullivan** [comment]
Fix reverted in #2052
